lib/cisco_node_utils/bridge_domain.rb

```diff
@@ -0,0 +1,275 @@
+
+module Cisco
+  # node_utils class for bridge_domain
+  class BridgeDomain < NodeUtil
+    attr_reader :name, :bd_ids
+
+    def initialize(bds, instantiate=true)
+      # Spaces are removed as bridge-domain cli doesn't accept value with
+      # space
+      @bd_ids = bds.to_s.gsub(/\s+/, '')
+      fail 'bridge-domain value is empty' if @bd_ids.empty? # no empty strings
+
+      create if instantiate
+    end
+
+    # This will expand the string to a list of bds as integers
+    def self.bd_ids_to_array(bdid_string)
+      list = []
+      narray = bdid_string.split(',')
+      narray.each do |elem|
+        if elem.include?('-')
+          es = elem.gsub('-', '..')
+          ea = es.split('..').map { |d| Integer(d) }
+          er = ea[0]..ea[1]
+          list << er.to_a
+        else
+          list << elem.to_i
+        end
+      end
+      list.flatten
+    end
+
+    # This method will generate a batched string if a list is passed as
+    # argument
+    # Input would be as [1,2,3,4,5,10,11,12,7,30,100,31,32]
+    # output will be 1-5,10-12,7,30,100,31-32
+    def self.bd_list_to_string(bd_list)
+      farray = bd_list.compact
+      lranges = []
+      unless farray.empty?
+        left = bd_list.first
+        right = nil
+        farray.each do |aelem|
+          if right && aelem != right.succ
+            if left == right
+              lranges << left
+            else
+              lranges << Range.new(left, right)
+            end
+            left = aelem
+          end
+          right = aelem
+        end
+        if left == right
+          lranges << left
+        else
+          lranges << Range.new(left, right)
+        end
+      end
+      lranges.to_s.gsub('..', '-').delete('[').delete(']').delete(' ')
+    end
+
+    def self.allbds
+      hash = {}
+      bd_list = config_get('bridge_domain', 'all_bds')
+      return hash if bd_list.nil?
+
+      final_bd_list =
+          bd_list.map { |elem| BridgeDomain.bd_ids_to_array(elem) }.flatten.uniq.sort
+      final_bd_list.each do |id|
+        hash[id] = BridgeDomain.new(id, false)
+      end
+      hash
+    end
+
+    # This function will first add bds to the system bridge-domain and then
+    # create the bds. If bds already existing then just create. Else add the
+    # non added bds to system range first then create all. This is to avoid the
+    # idempotency issue as system add command throws error if a bd is already
+    # present in the system range.
+    def create
+      sys_bds_array = BridgeDomain.bd_ids_to_array(system_bd_add)
+      inp_bds_array = BridgeDomain.bd_ids_to_array(@bd_ids)
+      if (inp_bds_array - sys_bds_array).any?
+        add_bds = BridgeDomain.bd_list_to_string(inp_bds_array - sys_bds_array)
+        config_set('bridge_domain', 'system_bd_add', addbd: add_bds)
+      end
+      result = config_set('bridge_domain', 'create', crbd: @bd_ids)
+      cli_error_check(result)
+    rescue CliError => e
+      bd_raise(e)
```

Comment:
@glennmatthews Committed a code that will alllow you to remove these `rescue` blocks.

See #266